docs/adr/0012_construct_as_transformation.md

```diff
@@ -0,0 +1,441 @@
+  avoid problems in complex graphs.
+
+## Comparison with Transformation ADR
+- Instead of expecting a component descriptor as a starting point for all 
+  operations that is consecutively modified and then merged (to detect 
+  conflicts), we create a new component descriptor to be uploaded 
+  with the resources created in the previous operations.
+- This simplifies the operations, as we do not have to deal with 
+  merging and conflict detection.
+- This would also be adopted for the transfer use case. Here, the component 
+  to be uploaded at the target location would be created from scratch and filled
+  with information from the original component descriptor through `cel` expressions.
+
+## Implementation
+
+### Input to Access Types
+
+As mentioned above, there will no longer be an implementation for each input 
+type. Instead, we want new `access types` (e.g. `file`, `dir`, ...) and 
+`uploaders`.
+
+### Plugin Registries
+
+To keep the complexity in check, we assume the transformation implementation 
+is aware of the available *capabilities*, their interfaces and their 
+types (or rather, schemas). This means, new *capabilities* can only be added by 
+extending the code.
+
+### Program Flow
+So, the logic will be heavily inspired by the current construct implementation.
+The overall flow will roughly be as follows:
+
+**1. Discover**
+- Discover loops over the entries in the (already parsed) transformation 
+  specification go struct. We do 2 iterations:
+  - **First Iteration**
+    - For each item in the `transformations` list:
+      - We create a vertex with the transformation `id` as the vertex `id`.
+  - **Second Iteration**
+    - For each item in the `transformations` list:
+      - We parse the cel expressions in the transformation `config` (without 
+        evaluating them yet). 
+      - We add edges based on the cel expressions in the `config` (add edge 
+        always does a cycle check). Here, we can error if the vertex with 
+        the `id` referenced in the cel expression does not exist because of 
+        the first iteration.
+      - We unmarshal the `config` into its corresponding go type. As 
+        mentioned above, we do not want an additional registry level. So, we 
+        do a switch case on the transformation `type` and perform the 
+        unmarshalling. Doing this here, instead of during process time,
+        allows us to error early if the config is invalid.
+      - We add the transformation with the unmarshalled `config` as an 
+        attribute to the vertex.
+
+**2. Process**
+- Process loops over the graph in topological order. For each vertex:
+  - We get the transformation with the typed `config` from the vertex attribute.
+  - We evaluate all cel expressions in the `config`, using the outputs of 
+    the predecessor vertices as input.
+  - We do another switch case statement on the transformation type to call the
```

Comment:
how do people know which transformation types are available and how they are used?